startables/startables.py

```diff
@@ -144,24 +153,17 @@ def __init__(self, df: pd.DataFrame, name: str, col_specs: Dict[str, ColumnMetad
 
         self._df = df
         self.remark = remark
-        if col_display_digits:
-            self._validate_col_display_digits(col_display_digits, df.columns)
-        self.col_display_digits = col_display_digits
 
     @staticmethod
     def _validate_col_specs(col_specs, df_col_names):
         missing_cols = [col_name for col_name in df_col_names if col_name not in col_specs]
         if missing_cols:
             raise ValueError(
                 f'Missing column specifications for DataFrame columns: {missing_cols}.')
-
-    @staticmethod
-    def _validate_col_display_digits(col_display_digits, df_col_names):
-        for col_name, digits in col_display_digits.items():
-            if col_name not in df_col_names:
-                raise ValueError(f'Column name not found in DataFrame columns: {col_name}')
-            if not isinstance(digits, int):
-                raise ValueError(f'Display digits value for DataFrame column {col_name} is not an integer.')
+        for col_name, spec in col_specs.items():
+            if spec.format_str:
+                if '{' not in spec.format_str or '}' not in spec.format_str:
+                    raise ValueError(f'Format string for column {col_name} not in correct format.')
```

Comment:
I suggest: allow user to either supply 
* a full `{:1.3f}` (with maybe additional stuff outside the `{}` such as `$` in your test) 
* or just a stripped down `1.3f`, which just gets auto wrapped in `{:...}` ?

i.e. any of these would be valid.
Would there be any undesirable complexity/side-effects to this... can't think of any at the moment